File: dsa-interview-crash-course/0344_reverse_string.cpp
#include <iostream>
#include <vector>

class Solution
{
public:
    void reverseString(std::vector<char> &s)
    {
        int l = 0;
        int r = s.size() - 1;
        while (l < r)
        {
            std::swap(s[l], s[r]);
            ++l;
            --r;
        }
    }
};

File: dsa-interview-crash-course/0344_reverse_string.rs
impl Solution {
    pub fn reverse_string(s: &mut Vec<char>) {
        let mut l = 0;
        let mut r = s.len() - 1;
        while l < r {
            s.swap(l, r);
            l += 1;
            r -= 1;
        }
    }
}
